components/layout_2020/flow/float.rs

```diff
@@ -939,23 +947,60 @@ impl SequentialLayoutState {
                 .max(self.floats.clear_right_position),
         };
         if hypothetical_block_position >= clear_position {
-            return None;
+            None
+        } else {
+            Some(clear_position)
         }
+    }
 
-        // We need to add clearance between `current_margin` and `block_start_margin`,
-        // this prevents them from collapsing.
-        // Compute the position of the element's top border edge if we added
-        // a clearance of 0px.
-        let position_with_zero_clearance = self.bfc_relative_block_position +
-            self.current_margin.solve() +
-            block_start_margin.solve();
-
-        // Set clearance to the amount necessary to place the border edge of the block
-        // even with the bottom outer edge of the lowest float that is to be cleared.
-        // Note that CSS2 also allows the alternative option of flooring this amount by
-        // `hypothetical_block_position - position_with_zero_clearance`,
-        // but other browser don't seem to do that.
-        Some(clear_position - position_with_zero_clearance)
+    /// Returns the amount of clearance (if any) that a block with the given `clear` value
+    /// needs to have at `current_block_position_including_margins()`.
+    /// `block_start_margin` is the top margin of the block, after collapsing (if possible)
+    /// with the margin of its contents. This must not be included in `current_margin`,
+    /// since adding clearance will prevent `current_margin` and `block_start_margin`
+    /// from collapsing together.
+    ///
+    /// https://www.w3.org/TR/2011/REC-CSS2-20110607/visuren.html#flow-control
```

Comment:
One thing I learned recently is that a raw link in a rustdoc comment is an error. You can fix this by writing:

`<https://www.w3.org/TR/2011/REC-CSS2-20110607/visuren.html#flow-control>`

instead.